### PR TITLE
Add chaincode events in commit notification

### PR DESCRIPTION
**Pull request by manish-sethi**
Signed-off-by: manish <manish.sethi@gmail.com>

#### Type of change
- Improvement 

#### Description

This PR includes chaincode event details in the commit events

### CODE REVIEW

File: core/ledger/ledger_interface.go [hunk 1]
```diff
@@ -733,9 +733,19 @@ type HashProvider interface {
...
+// CommitNotification is sent on each block commit to the channel returned by PeerLedger.CommitNotificationsChannel()
 type CommitNotification struct {
-	BlockNumber         uint64
-	TxIDValidationCodes map[string]peer.TxValidationCode
+	BlockNumber uint64
+	TxsByTxID   map[string]*CommitNotificationTxInfo
+}
+
+// CommitNotificationTxInfo contains the details of a transaction that is included in the CommitNotification
+// ChaincodeID will be nil if the transaction is not an endorser transaction. This may or may not be nil if the tranasction is invalid. However, it will always be non-nil if the transaction is valid.
```
Comment (denyeart) [inline core/ledger/ledger_interface.go]: Can you expand on "This may or may not be nil if the transaction is invalid".  What are the different scenarios?

**Comment by denyeart** [comment]
LGTM besides that one comment nit.

**Review by bestbeforetoday — DISMISSED** [review]
👍

File: core/ledger/ledger_interface.go (hunk 1 above)
Comment (manish-sethi) [reply core/ledger/ledger_interface.go]: Added more details in the comment

**Review by denyeart — APPROVED** [review]

**Pull request merged**